lib/OpenCloud/ObjectStore/Resource/DataObject.php

```diff
@@ -354,6 +383,60 @@ public function delete($params = array())
     {
         return $this->getService()->getClient()->delete($this->getUrl())->send();
     }
+    
+    /**
+     * @param string $destination Path (`container/object') of other object to symlink this object to
+     * @return null|\Guzzle\Http\Message\Response The response or null if $this is not empty
+     */
+    public function createSymlinkTo($destination)
+    {
+        if (!$this->name) {
+            throw new Exceptions\NoNameError(Lang::translate('Object has no name'));
+        }
+
+        if ($this->getContentLength()) {
+            throw new ObjectNotEmptyException($this->getContainer()->getName() . '/' . $this->getName());
+        }
+
+        $response = $this->getService()
+            ->getClient()
+            ->createRequest('PUT', $this->getUrl(), array(
+                HeaderConst::X_OBJECT_MANIFEST => (string) $destination
+            ))
+            ->send();
+
+        if ($response->getStatusCode() == 201) {
+            $this->setManifest($source);
+        }
+
+        return $response;
+    }
+
+    /**
+     * @param string $source Path (`container/object') of other object to symlink this object from
+     * @return null|DataObject The symlinked object or null if object already exists and is not empty
```

Comment:
Hey @markchalloner, could you please remove the `null` from the @return type and description? Also, could you please add the necessary [@throws](http://phpdoc.org/docs/latest/references/phpdoc/tags/throws.html) documentation for the exceptions thrown by this method?